src/utils/requireOptionalPeer.ts

```diff
@@ -0,0 +1,27 @@
+import { createRequire } from 'module'
+
+// @ts-ignore - cjs build will complain about import.meta not being allowed, but its not actually used as __filename will be defined for cjs
+const file = typeof __filename !== 'undefined' ? __filename : import.meta.url
+const require = createRequire(file)
```

Comment:
I opted for using a require here as its behavior is synchronous and I believe the functions where these were being called from were also synchronous and not returning promises. Ideally, we would use a dynamic `import` for the esm code, and a `require` for the cjs code, but introducing the dynamic import would change the signatures of these functions. Let me know if you would like me to pursue this.

Reply:
I am not sure what the best way to proceed here is, but this seems really heavy handed to make a breaking change and add all this complexity to support one specific package manager.

Reply:
Functionally the require method being used should be fine and match existing behavior.